### ISSUE TITLE
mismatch `veclength` between `Zygote.gradient` and `Flux.params` 

### ISSUE THREAD

**Issue by Red-Portal** [baggepinnen/FluxOptTools.jl#9]
Hi, it seems `veclength` has a bug where the length of the gradient and `Flux.params` yields different values.
This triggers an internal assertion macro.
Here's a reproducible example. 

```julia
using Bijectors
using Flux
using FluxOptTools
using Optim
using StatsBase
using Zygote

    data = randn(2, 10)
    base_dist = MvNormal(zeros(2), ones(2))
    layers    = reduce(∘, [
        PlanarLayer(2)
        for i in 1:4
    ])
    flow = transformed(base_dist, layers)
    loss = () -> -sum(logpdf.(Ref(flow), eachcol(data)))
    Zygote.refresh()
    pars = Flux.params(flow)

    grads = Zygote.gradient(loss, pars)

    println(veclength(pars))
    println(veclength(grads))

   @assert veclength(pars) == veclength(grads)

```
